apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/PartitionedReadTests.cs

```diff
@@ -29,9 +31,52 @@ public class PartitionedReadTests
         public PartitionedReadTests(PartitionedReadTableFixture fixture) =>
             _fixture = fixture;
 
-        [CombinatorialData]
-        [Theory]
-        public async Task DistributedReadAsync(bool query)
+        [Theory, CombinatorialData]
+        [Trait(Constants.SupportedOnEmulator, Constants.No)]
+        public async Task DistributedReadAsync(bool query, bool dataBoostEnabled)
+        {
+            int numRows;
+            using (var connection = _fixture.GetConnection())
+            {
+                using (var cmd = connection.CreateSelectCommand($"SELECT COUNT(*) FROM {_fixture.TableName}"))
+                {
+                    numRows = await cmd.ExecuteScalarAsync<int>();
+                }
+            }
+
+            using (var connection = new SpannerConnection(_fixture.ConnectionString))
+            {
+                await connection.OpenAsync();
+
+                using (var transaction = await connection.BeginReadOnlyTransactionAsync())
+                using (var cmd = query
+                    ? connection.CreateSelectCommand($"SELECT * FROM {_fixture.TableName}")
+                    : connection.CreateReadCommand(_fixture.TableName, ReadOptions.FromColumns(_fixture.ColumnNames), KeySet.All))
+                {
+                    transaction.DisposeBehavior = DisposeBehavior.CloseResources;
+                    cmd.Transaction = transaction;
+                    var partitions = await cmd.GetReaderPartitionsAsync(PartitionOptions.Default.WithPartitionSizeBytes(100).WithDataBoostEnabled(dataBoostEnabled));
+                    var transactionId = transaction.TransactionId;
+
+                    var serDserPartition = CommandPartition.FromBase64String(partitions.FirstOrDefault().ToBase64String());
+                    var request = serDserPartition.Request.ExecuteSqlRequest;// TODO add ternary operator
+                    Assert.Equal(request.DataBoostEnabled, dataBoostEnabled);
+
+
+                    //we simulate a serialization/deserialization step in the call to the subtask.
+                    await Task.WhenAll(partitions.Select(
+                            x => DistributedReadWorkerAsync(CommandPartition.FromBase64String(x.ToBase64String()),
```

Comment:
Again, this code would be much simpler without the serialization/deserialization - which is easy to unit test separately.